### ISSUE TITLE
Useless translation of 'onready' and 'ontimeout' in soundmanager2-jsmin.js

### ISSUE THREAD

**Issue by popov-d** [owncloud/music#671]
File js/vendor/soundmanager/script/soundmanager2-jsmin.js is a minified version, not an initial source. Function names are also minified, and one of them got a name of t(). Unfortunately this file got into list of translated files, so Transiflex thinks that t() is a call for translation. Consequently, arguments 'onready' and 'ontimeout' appear in Transiflex as "strings needed to be translated", while in fact these seem to be JavaScript event names. Obviously it's wrong, no user will ever see these "translations".

I think that minified files should somehow be excluded from translation process (only real sources should be included when needed).

P. S. I am not a developer, a sysadmin with programming knowledge, just a translator to Russian in this case.

**Comment by paulijar** [comment]
Yes, Transifex is currently extracting localization strings of the Music app incorrectly. These two extraneous strings is just the top of the iceberg, the bigger problem is that some 80 % of the translatable strings of the app are missing from Transifex. For this reason, we can't get any new translations to the application at the moment.

This is something we cannot fix on the application side (alone) but we need support from the ownCloud core team. The issue is tracked at https://central.owncloud.org/t/owncloud-music-app-translations/14881 but unfortunately there has been no progress during the last two months.

**Comment by TtuxX** [comment]
Hi @paulijar ! :)

I have seen that large portions of the Music app are still not translated and would love to actively contribute and help translate the Music app (to French, for my part).

I've seen your last reply on the ownCloud forum thread (https://central.owncloud.org/t/owncloud-music-app-translations/14881/14) and, using Nextcloud myself, I do think too that it would be a great idea to move the Music app over to Nextcloud.

Following your message on issue https://github.com/owncloud/music/issues/712, do you think we could now make the choice to move the app to Nextcloud and ask ownCloud team to actively help us so that we can catch-up again with the translations ?

Thank you in advance for your help ! :)

**Comment by paulijar** [comment]
@TtuxX Yes, one way or another, it would probably be the best move to migrate this project to Nextcloud. Based on all the reported issues, my impression is that something like 80-90% of the userbase of the app are nowadays on Nextcloud anyway.

The inhibiting factor here is that this github project is owned by ownCloud, and I can't just move it without their help. I'm not sure if they are willing to help here, but at least they don't seem to have been terrible eager when I have tried to ask or hint about it. I have mostly got silence as answer. Maybe it would help if I would push harder, but so far I have always had something more urgent or interesting to put may time and energy on.

If no help is got from ownCloud, then it would still be possible to create a new fork of the project on Nextcloud. But the downside on this would be that the open and closed issues and PRs wouldn't be transferred to the new project. And another problem would be, that then there would be two projects in github, causing confusion about where the issues and feature requests should be reported.

**Comment by TtuxX** [comment]
Thank you very much for your detailed answer Pauli.
Arf... you are right this issue is not easy to solve and forking the project while loosing all the history, issues and PRs is not the best solution.
Would it be possible to integrate translations somehow through Github directly ? Or we cannot do much and thus cannot expect to be able to solve the issue anytime soon by any possible mean ? :/

**Comment by paulijar** [comment]
> Would it be possible to integrate translations somehow through Github directly ?

This is actually a good idea to get some kind of solution right now. I made some investigation, and this is how the process could work:

1. Pull/fork the Music project from github
2. Download the Poedit application from here: https://poedit.net/
3. Start Poedit, and open the *.po file matching the language you wish to modify. These files can be found under the Music repository from paths like `l10n/<langcode>/music.po`.
4. From the menus of Poedit, select to update the catalog from POT file. Browse and select the file `l10n/templates/music.pot`
    * This imports all the latest translation placeholders to your language-specific .po file
5. Add/update all the translations you wish and then save the file.
    * Especially, check/update all the translations highlighted with orange, as these have been somehow heuristically evaluated by Poedit and are often wrong.
6. Create a pull request with your modified .po file

If you would like to actually test the updated translations, then you will need to navigate to the folder `build` under the music repository, and give commands
`make l10n-compile`
`make`

The first of the commands will gather translations from all the *.po files to the file `js/app/l10n/translations.js` and the second will create the webpack bundle to be deployed on your server. Running these commands will require you to have at least the `make` and `npm` utilities. If you do have these utilities installed, then you may include also the modified translations.js file in your PR. But if not, then I can always regenerate it later.
